### PR TITLE
Create unlock transaction asset for dpos module - Partially Closes #5608

### LINKED ISSUE #5608: Create DPoS module
### Description

* [x]  Create RegisterAsset which extends BaseAsset
    *   Name should be “register”
    *   Type should be 0
    *   Move asset schema from delegate transaction
    *   Move validateAsset functionality and tests from Delegate transaction
    *   Move applyAsset functionality and tests from Delegate transaction
* [x]    Create VoteAsset which extends BaseAsset
    *   Name should be “vote”
    *   Type should be 1
    *   Move asset schema from vote transaction
    *   Move validateAsset functionality and tests from Vote transaction
    *   Move applyAsset functionality and tests from Vote transaction
    *   Update change of balance to use reducer “token:debit” to lock the balance from the sender
*  [x]   Create UnlockAsset which extends BaseAsset
    *   Name should be “unlock”
    *   Type should be 2
    *   Move asset schema from unlock transaction
    *   Move validateAsset functionality and tests from unlock transaction
    *   Move applyAsset functionality and tests from unlock transaction
    *   Update change of balance to use reducer “token:credit” to return the balance to the sender
*  [x]   Create PoMAsset which extends BaseAsset
    *   Name should be “pom”
    *   Type should be 3
    *   Move asset schema from pom transaction
    *   Move validateAsset functionality and tests from pom transaction
    *   Move applyAsset functionality and tests from pom transaction
*  [x]   Create DPoSModule which extends BaseModule
    *   Move accountSchema “asset” from framework to accountSchema
    *   Name should be “dpos”
    *   Type should be 5
    *   Add validation of genesis config in the constructor
        *   Check activeDelegates, standbyDelegates and delegateListRoundOffset exist and proper type
    *   Move dpos.onBlockFinalized functionality to afterBlockApply step
        *   forgerList will not be managed in DPoS, so not required to be moved
        *   Use consensus.getFinalizedHeight to get the finalized height for cleanup
    *   Move genesis block validation related to DPoS from lisk-genesis to afterGenesisBlockApply
    *   Move updateProductivity logic from dpos/delegate_info to afterBlockApply step
    *   Move applyBootstrap logic from dpos/delegate_info to afterBlockApply step
        *   During the bootstrap period, it should not updateValidator
        *   Only at the end of the bootstrapPeriod, which should be end of the round, should call **updateValidator**
        *   It should still snapshot the voteWeight at the end of the round
    *   Move update logic from dpos/delegate_info to afterBlockApply step
        *   Standby delegate should have canVote false, else true
        *   Instead of calling setForgerList, it should call **updateValidator**

#### General concept

- afterGenesisBlockApply
   * Validate dpos related properties in genesis account
- afterBlockApply
   * Update delegate productivity
   * Snapshot voteWeight at the end of the round
   * Skip updateValidators during bootstrap period except last block of the period
   * Using the snapshot voteWeight, create validators list and call updateValidators
   * Cleanup finalized state data

### Motivation

DPoS module is a module to handle DPoS selection protocol

### Acceptance criteria

*   DPoS module should not be dependent on lisk-dpos (it will be removed)
*   All related unit tests are moved to module/dpos unit test

### Related Issues

#5603

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #5608

### How was it solved?

Added the transaction asset for unlock transaction.

### How was it tested?

Run all unit tests.

### CODE REVIEW

File: framework/test/unit/specs/modules/dpos/transaction_assets/unlock_transaction_asset.spec.ts [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+
+				// Assert
+				const updatedSender = await stateStoreMock.account.get<Account<DPOSAccountProps>>(
+					sender.address,
+				);
+
+				expect(updatedSender.dpos.unlocking).toHaveLength(0);
+			});
+		});
+
```
Comment (shuse2) [inline framework/test/unit/specs/modules/dpos/transaction_assets/unlock_transaction_asset.spec.ts:628]: > when account contain duplicate unlocking entries but asset.unlockObjects only contains one

This test is missing?

File: framework/test/utils/assets.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright © 2019 Lisk Foundation
```
Comment (shuse2) [inline framework/test/utils/assets.ts:2]:
```suggestion
 * Copyright © 2020 Lisk Foundation
```

File: framework/test/utils/node/state_store_mock.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ import { createFakeDefaultAccount } from './account';
 import { AccountStateStore, Account, AccountDefaultProps } from '../../../src/modules/base_asset';
 
 export interface ChainState {
-	readonly lastBlockHeaders: BlockHeader[];
+	lastBlockHeaders: BlockHeader[];
```
Comment (shuse2) [inline framework/test/utils/node/state_store_mock.ts:21]: this shouldn't be modified

Comment (nazarhussain) [reply framework/test/utils/node/state_store_mock.ts:21]: I can add a comment there for reference, there are a lot of use cases in tests to change the last block headers, if we make it readonly, make it redundant to re-initialize the whole StateStoreMock object.

**Pull request merged**